src/main/java/com/gluonhq/substrate/SubstrateDispatcher.java

```diff
@@ -248,4 +251,33 @@ private static String prepareDirs(Path buildRoot) throws IOException {
 
     }
 
+    /**
+     * check if the GraalVM provided by the configuration is capable of running native-image
+     * @param configuration
+     * @throws NullPointerException when the configuration is null
+     * @throws IllegalArgumentException when the configuration doesn't contain a property graalPath
+     * @throws IOException when the path to bin/native-image doesn't exist
+     */
+    static void assertGraal(ProjectConfiguration configuration) throws IOException {
+        Objects.requireNonNull(configuration);
+        String graalPathString = configuration.getGraalPath();
+        if (graalPathString == null) throw new IllegalArgumentException("There is no GraalVM in the projectConfiguration");
+        Path graalPath = Path.of(graalPathString);
+        if (!graalPath.toFile().exists()) throw new IOException ("Path provided for GraalVM doesn't exist: "+graalPathString);
+        Path binPath = graalPath.resolve("bin");
+        if (!binPath.toFile().exists()) throw new IOException("Path provided for GraalVM doesn't contain a bin directory: "+graalPathString);
+        Path niPath = binPath.resolve("native-image");
+        if (!niPath.toFile().exists()) throw new IOException ("Path provided for GraalVM doesn't contain bin/native-image: "+graalPathString);
+        Path javacmd = binPath.resolve("java");
+        ProcessBuilder processBuilder = new ProcessBuilder(javacmd.toFile().getAbsolutePath());
+        processBuilder.command().add("-version");
+        processBuilder.redirectErrorStream(true);
+        Process process = processBuilder.start();
+        InputStream is = process.getInputStream();
+        BufferedReader br = new BufferedReader(new InputStreamReader(is));
+        String l = br.readLine();
+        if (l == null) throw new IllegalArgumentException("java -version failed to return a value for GraalVM in "+graalPathString);
+        if (l.indexOf("\"1.8") > 0) throw new IllegalArgumentException("You are using an old version of GraalVM in "+graalPathString+
```

Comment:
Do we need `\"`?

Reply:
For some reasons, the version on 8 is inside quotes (at least from GraalVM 19.2.x) , but not on 11. A reason to do it is to allow e.g. 11.8.0 (which is unlikely to occur).
But in the short term, it might indeed be better to remove the quotes, in case an 8 version is used that is not using quotes.